Release Notes and misc improvements and corrections (#2385)
tinymce/tinymce-docs: modules/ROOT/pages/6.0-release-notes-premium-changes.adoc

|===
| Old name                            | New name                         | Element                             | Notes

> | `formatpainter_blacklisted_formats` | `formatpainter_ignored_formats`  | Option                              | NB: _Image Tools_ is now _Enhanced Image Editing_, a https://tiny.cloud/tinymce/features/#productivity[Premium plugin].

Reviewer: This note is incorrect?
```suggestion
| `formatpainter_blacklisted_formats` | `formatpainter_ignored_formats`  | Option                              | 
```

| `toc_header`                        | `tableofcontents_header`         | Option                              | NB: _Table of Contents_ is now a https://tiny.cloud/tinymce/features/#productivity[Premium plugin].
|===

[[hindi-malay-and-vietnamese-translations-added-to-premium-plugins]]
== Hindi, Malay, and Vietnamese translations added to Premium Plugins

The following Premium Plugins now present in Hindi, Malay, and Vietnamese:

* Accessibility Checker
* Advanced Tables
* Comments
* Enhanced Image Editing
* Export
* PowerPaste
* Real-time Collaboration (RTC)
* Spell Checker Pro
* Table of Contents

// to do above
// link out to each plugin page
//
// pre-requisite

> // xref or https as link syntax?

Reviewer 2: xref links should be used within docs, unless we are linking outside of the docs section (https for tiny website links)

Author: xrefs added.

Previously tables containing a `colgroup` did not sort correctly. As of this release, this error has been corrected: tables with a `colgroup` now sort correctly, as expected.

=== Comments

The following changes and fixes were applied to the xref:introduction-to-tiny-comments.adoc[Comments] plugin:

Because of how a comment’s active state was captured when the comment was made active, this active state was not excluded when capturing editor content from a comment.

With this release, a comment’s active state is captured differently and, consequently, this state is excluded when comment content is captured.

Previously, every time the insertion point was placed within an extant comment and then placed anywhere outside of the comment body, an undo level was added. That is, every time focus was switched either to or from a content, an undo level was added.

With this update, shifting focus to or from a comment no longer adds undo levels, as expected.

Previously, the Comments plugin’s scroll position logic lead to, in some edge cases, the last comment not scrolling into view correctly.

The logic has been corrected and, with this update, all comments scroll into view as expected.

> Previously, when a Comment was added to a {productname} instance, the data in `tinycomments` was automatically, and incorrectly, added to the *Find and Replace* Find text-entry input field.

Reviewer 3: This isn't completely correct... it was incorrectly added to `editor.selection.getContent()` which meant it showed in dialogs, such as Find and Replace and Insert Link.

> Previously, when a Comment was added to a {productname} instance, the data in `tinycomments` was automatically, and incorrectly, added to the *Find and Replace* Find text-entry input field.

Reviewer 3: As a side note, this also showed in other dialogs such as the Insert link dialog.

Author: Edited to reflect the above side note.

The following premium plugin updates were released alongside {productname} 6.0.

=== Accessibility Checker

The default value for `a11ychecker_html_version` was `html4`.

It is now `html5`.

=== Advanced Table

The `getRowType` function would sometimes incorrectly parse elements in the `tfoot` section of a table.

> This led to `mceTableRowType` being incorrectly returned as a header if all cells in the section were `th` elements.

Reviewer 4: @ltrouton Could you please check this bit? It reads a bit odd to me, and you're the expert.

Reviewer 5: @brian-forte The information in the `Advanced Table` subsection isn't quite correct so would have to be reworked if we wanted to keep it. 

I am curious why this subsection was added as this is more of a basic fix and there were other fixes/changes made for the `advtable` plugin as well as part of 6.0. Apologies if this information is somewhere else that I haven't seen.

Author: > The information in the Advanced Table subsection isn’t quite correct
> so would have to be reworked if we wanted to keep it.

What’s the nature of the error in the current text?

> I am curious why this subsection was added as this is more of a
> basic fix and there were other fixes/changes made for the `advtable`
> plugin as well as part of 6.0.

It’s the only change to `advtable` that I had any info on (via [this portion](https://www.notion.so/tinycloud/6-0-Docs-Notes-94ee2ecf10ff440b9246a22f8cc54828#3e4cd38137354dcd8fbe92d7dc7c5eaf) of [this Notion](https://www.notion.so/tinycloud/6-0-Docs-Notes-94ee2ecf10ff440b9246a22f8cc54828) doc).

Author: Above questions answered via Slack DM.

With this update, moving the insertion point through a Horizontal Rule using arrow keys works as expected: the insertion point moves and no exception is thrown.

Previously, attempting to remove failed image uploads with the `images_upload_handler` API failed if Real-time Collaboration was loaded.

With this update, `images_upload_handler` removes failed image uploads and presents the correct alert message.

Keyboard events can now be prevented before the Real-time Collaboration plugin intercepts them.

Previously, selecting linked images when the `link_context_toolbar` option was enabled, caused the Real-time Collaboration editor to enter an endless loop. This loop caused the selected images to flicker.

With this update, the Real-time Collaboration editor no longer enters an endless loop when linked images are selected. Consequenlty, the selected images no longer flicker.

=== Spell Checker Pro

The free `spellchecker` plugin was removed as no longer supported.

> The https://tiny.cloud/tinymce/features/#productivity[Premium plugin], xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro], which offers equivalent functionality and more, is available.

Reviewer 3: This should be using the existing AsciiDoc attributes, e.g:
```suggestion
The link:{plugindirectory}#productivity[Premium plugin], xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro], which offers equivalent functionality and more, is available.
```

Author: This line has been excised completely.

The following fixes were made to the xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro] plugin:

Previously, Spell Checker Pro treated soft hyphens (ie the `+&shy;+` html entity) as a word boundary. As a consequence words containing soft hyphens were not correctly spellchecked as whole words.

> With this update, `+&shy;+` entity is correctly recognised as part of a word, not a word boundary. And words containing soft hyphens are, consequently, spell checked correctly.

Reviewer 3: ```suggestion
With this update, the `+&shy;+` entity is correctly recognised as part of a word, not a word boundary. And words containing soft hyphens are, consequently, spell checked correctly.
```

As well, one command and one option have been replaced and a, previously deprecated, fallback option has been removed.

[cols="1,1,1"]
|===
| Removed item                    | Replacement           | Element type

| `mceSpellcheckDocument`         | `mceSpellcheckDialog` | Command

| `spellchecker_select_languages` | `content_langs`       | Option

> | `spellchecker_whitelist`        | No replacement        | Option

Reviewer 3: This is incorrect, the replacement is `spellchecker_ignore_list` (or at least it's the same scenario as `content_langs`).

Author: Corrected.

If a community-maintained version of the `paste` plugin becomes available, we will link to it from the xref:migration-from-5x.adoc#plugins-paste[_Migration Guide_].

=== Real-time Collaboration

The following improvements, changes, and fixes were made to Real-time Collaboration.

Three additional options are now supported when Real-time Collaboration is active:

* `valid_elements`
* `valid_children`
* `extended_valid_elements`

Real-time Collaboration now supports the `autocompleter()` API.

> //xref:autocompleter.adoc[`autocompleter()`] ?

Reviewer 2: @TheSpyder can you confirm these links and the inline boundaries below

Reviewer: Neither of those references exist. I'll add some suggestions.

Author: After some discussion elsewhere in this PR, existing references and a ‘better approach’ was the way forward.

> Real-time Collaboration now supports the `autocompleter()` API.
> 
> //xref:autocompleter.adoc[`autocompleter()`] ?
> 
> Real-time Collaboration now supports the `inline_boundaries` option.
> 
> // xref:partial$configuration/inline-boundaries.adoc[`inline_boundaries`] ?

Reviewer: This isn't ideal — the docs stylesheet doesn't show links in code blocks other than on hover — but these are the links I think we should use.

```suggestion
Real-time Collaboration now supports the xref:autocompleter.adoc[`autocompleter`] API.

Real-time Collaboration now supports the xref:content-behavior-options.adoc#inline_boundaries[`inline_boundaries`] option.
```

Author: In terms of making the links more visible, would something like this work better?

+ Real-time Collaboration now supports xref:autocompleter.adoc[the `autocompleter` API].
+ 
+ Real-time Collaboration now supports xref:content-behavior-options.adoc#inline_boundaries[the `inline_boundaries` option].

Reviewer: yeah that would be better :)

Author: Going with the better approach. :-)

> When provider values are incorrect, the returned error messages are now less opaque.

Reviewer 6: ```suggestion
When provided values are incorrect, the returned error messages are now less opaque.
```

For example, the error message `'key' was not a string` is now the error message `Expected field 'key'`.

> Previously, if a {productname} editor instance with Real-time Collaboration loaded contained a Horizontal Rule (ie an `+<hr>+` element), attempting to move the insertion point over the Horizontal Rule using arrow keys threw an exception: `Uncaught Error: Unsupported node type: [object DocumentType]`.

Reviewer 6: ```suggestion
Previously, if a {productname} editor instance with Real-time Collaboration loaded and contained a Horizontal Rule (ie an `+<hr>+` element), attempting to move the insertion point over the Horizontal Rule using arrow keys threw an exception: `Uncaught Error: Unsupported node type: [object DocumentType]`.
```

Author: To quote my git commit comment:

Conjunction — also — added to sentence regarding HR bug fix. A different conjunction — and — was suggested in the PR but, on reflection, the potential for reader confusion is obviated more effectively by adding ‘also’ than by adding ‘and’.

With this update, moving the insertion point through a Horizontal Rule using arrow keys works as expected: the insertion point moves and no exception is thrown.

Previously, attempting to remove failed image uploads with the `images_upload_handler` API failed if Real-time Collaboration was loaded.

With this update, `images_upload_handler` removes failed image uploads and presents the correct alert message.

Keyboard events can now be prevented before the Real-time Collaboration plugin intercepts them.

Previously, selecting linked images when the `link_context_toolbar` option was enabled, caused the Real-time Collaboration editor to enter an endless loop. This loop caused the selected images to flicker.

With this update, the Real-time Collaboration editor no longer enters an endless loop when linked images are selected. Consequenlty, the selected images no longer flicker.

=== Spell Checker Pro

> The free `spellchecker` plugin was removed as no longer supported.

Reviewer: I'm not sure we really need to note that in the premium docs 🤔 it doesn't impact spell checker pro at all.

Author: It’s mentioned primarily for technical marketing purposes.

The sub-text is _you were using the free spellchecker? Sorry, it’s now gone. This is the alternative._

Reviewer: I can appreciate that in the core docs, but I'm not sure a free user will read the premium changes doc.

Reviewer 3: I'd agree with Andy here, this makes little sense to be in a section about premium plugins.

Author: The consensus is in, and the above Sentence has been excised. (As well as the following sentence, which didn’t make sense once the first sentence was excised.)

Users of the Free version of {productname} 5.x who rely on the, now deprecated and removed, `toc` plugin, should not upgrade to {productname} 6.0 without a https://tiny.cloud/pricing/[subscription plan] that provides access to Premium plugins such as https://tiny.cloud/tinymce/features/#productivity[_Table of Contents_].

The new Premium `tableofcontents` plugin also changes the names of several elements.

> [cols="1,1"]

Reviewer: Perhaps add `element type` here as you did with spelling - there are a few element types in this list as listed at the top of the file.

Author: `Element type` column added. Two questions emerged when putting the new column together.

1. What is `tableofcontentsupdate`?

2. Are the other elements correct?

Reviewer: It's a toolbar item, this is already listed at the top of the file (line 43) :)

Author: ::facepalm::

Also: thanks for pointing me to my own work. It’s not less mortifying, but it is appropriately mortifying. :-)

The {productname} 6.0 release includes accompanying changes affecting the {productname} *self-hosted* services for the following plugins:

* The xref:introduction-to-mediaembed.adoc[Enhanced Media Embed] plugin (`+mediaembed+`)
* The xref:export.adoc[Export] plugin (`+export+`)
* The xref:editimage.adoc[Enhanced Image Editing] plugin (`+editimage+`)
* The xref:linkchecker.adoc[Link Checker] plugin (`+linkchecker+`)

> * The xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro] plugin (`+tinymcespellchecker+`)

Reviewer 3: This shouldn't have been changed, as the links are just below this instead. As such it's quite odd to have them linked twice.

If we do want this changed though, then the template in https://github.com/tinymce/tinymce-docs/blob/staging/docs-6/_new_content_templates/releasenotes_blank/x_y_z-release-notes-premium-changes.adoc also needs changing.

Reviewer 4: Done - removed "The" and template file was also updated.
